pkg/redisnode/node.go

```diff
@@ -52,33 +53,49 @@ func (n *Node) Clear() {
 
 // UpdateNodeConfigFile update the redis config file with node information: ip, port
 func (n *Node) UpdateNodeConfigFile() error {
-	err := n.addSettingInConfigFile("port " + n.config.RedisServerPort)
-	if err != nil {
+	// import default configuration if the config file was not the default value
+	if n.config.Redis.ConfigFileName != config.RedisConfigFileDefault {
+		if err := n.addSettingInConfigFile("include " + config.RedisConfigFileDefault); err != nil {
+			return err
+		}
+	}
+
+	if err := n.addSettingInConfigFile("port " + n.config.Redis.ServerPort); err != nil {
 		return err
 	}
-	if n.config.RedisMaxMemory > 0 {
-		err = n.addSettingInConfigFile(fmt.Sprintf("maxmemory %d", n.config.RedisMaxMemory))
-		if err != nil {
+
+	if err := n.addSettingInConfigFile("cluster-enabled yes"); err != nil {
+		return err
+	}
+
+	if n.config.Redis.MaxMemory > 0 {
+		if err := n.addSettingInConfigFile(fmt.Sprintf("maxmemory %d", n.config.Redis.MaxMemory)); err != nil {
 			return err
 		}
 	}
-	if n.config.RedisMaxMemoryPolicy != RedisMaxMemoryPolicyDefault {
-		err = n.addSettingInConfigFile(fmt.Sprintf("maxmemory-policy %s", n.config.RedisMaxMemoryPolicy))
-		if err != nil {
+	if n.config.Redis.MaxMemoryPolicy != config.RedisMaxMemoryPolicyDefault {
+		if err := n.addSettingInConfigFile(fmt.Sprintf("maxmemory-policy %s", n.config.Redis.MaxMemoryPolicy)); err != nil {
 			return err
 		}
 	}
-	err = n.addSettingInConfigFile("bind " + n.IP + " 127.0.0.1")
-	if err != nil {
+
+	if err := n.addSettingInConfigFile("bind " + n.IP + " 127.0.0.1"); err != nil {
 		return err
 	}
-	err = n.addSettingInConfigFile("cluster-node-timeout " + strconv.Itoa(n.config.Redis.ClusterNodeTimeout))
-	if err != nil {
+
+	if err := n.addSettingInConfigFile("cluster-node-timeout " + strconv.Itoa(n.config.Redis.ClusterNodeTimeout)); err != nil {
 		return err
 	}
 	if n.config.Redis.GetRenameCommandsFile() != "" {
-		err = n.addSettingInConfigFile("include " + n.config.Redis.GetRenameCommandsFile())
-		if err != nil {
+
+		if err := n.addSettingInConfigFile("include " + n.config.Redis.GetRenameCommandsFile()); err != nil {
+			return err
```

Comment:
In general in all these `return err` I would context... `fmt.Errorf("unable to set cluster-node-timeout in config file: %v", err)` or similar.
Trying to imrpve a little the UX

Reply:
right, I will add error context in the function `addSettingInConfigFile` since it is a local function and we call it 10times.

Reply:
cool, thanks